app/services/teacher_training_public_api/sync_sites.rb

```diff
@@ -58,12 +68,13 @@ def study_modes(course)
       (from_existing_course_options + [course.study_mode]).uniq
     end
 
-    def create_course_options(site, study_mode, site_status)
+    def create_course_options(site, temp_site, study_mode, site_status)
       course_option = CourseOption.find_or_initialize_by(
         site: site,
         course_id: course.id,
         study_mode: study_mode,
       )
+      course_option.update(temp_site: temp_site)
```

Comment:
could we move this to the end of this method, because if we have a new course option, that we haven't set the vacancy status for, and attempt to update here we will get an error as the `vacancy_status` should never be `nil`. This seems to work now anyway as when we get to the `update` vacancy status for new records we are persisting the `temp_site` then, let me know your thoughts

Reply:
🆗